sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -240,8 +240,31 @@ abstract class JdbcDialect extends Serializable with Logging{
         getJDBCType(dataType).map(_.databaseTypeDefinition).getOrElse(dataType.typeName)
       s"CAST($l AS $databaseTypeDefinition)"
     }
+
+    override def visitSQLFunction(funcName: String, inputs: Array[String]): String = {
+      if (isSupportedFunction(funcName)) {
+        s"""$funcName(${inputs.mkString(", ")})"""
+      } else {
+        throw QueryCompilationErrors.noSuchFunctionError(dialectName, funcName)
+      }
+    }
   }
 
+  /**
+   * Returns whether the database supports function.
+   * @param funcName Function name
+   * @return True if the database supports function.
+   */
+  @Since("3.3.0")
+  def isSupportedFunction(funcName: String): Boolean = false
+
+  /**
+   * Returns the name of database dialect.
+   * @return
+   */
+  @Since("3.3.0")
+  def dialectName: String = this.getClass.getCanonicalName
```

Comment:
Do we really need an API for this? I think the caller side can just write `dialect.getClass.getSimpleName`